utils/install-dependencies.sh

```diff
@@ -81,12 +81,6 @@ function install_dependencies_with_apt() {
     sudo apt-get install -y curl make gcc g++ cpanminus libpcre3 libpcre3-dev libldap2-dev unzip openresty-zlib-dev openresty-pcre-dev
 }
 
-# Install dependencies on mac osx
-function install_dependencies_on_mac_osx() {
-    # install OpenResty, etcd and some compilation tools
-    brew install openresty/brew/openresty luarocks lua@5.1 wget curl git pcre openldap
```

Comment:
I think we can throw an warning message, it more friendly

Reply:
hi, i have update this, in fact, if it is Mac in this logic, it will indicate that the current platform is not supported, so I did not add a new prompt. 
<img width="1097" alt="image" src="https://github.com/apache/apisix/assets/19872346/8cf478b9-3da3-4e04-b1f8-427d93d91afc">

But there is another problem with this code. Even the unsupported platform `make deps` will continue to execute, so I added an `exit 1` to interrupt it.
<img width="1109" alt="image" src="https://github.com/apache/apisix/assets/19872346/b8f05258-e7e4-4870-8c87-b7b212e051bb">